cpp/src/parquet/file_writer.cc

```diff
@@ -140,38 +143,39 @@ class RowGroupSerializer : public RowGroupWriter::Contents {
 
     const int32_t column_ordinal = next_column_index_++;
     const auto& path = col_meta->descr()->path();
+    const ColumnProperties& column_properties = properties_->column_properties(path);
     auto meta_encryptor =
         file_encryptor_ ? file_encryptor_->GetColumnMetaEncryptor(path->ToDotString())
                         : nullptr;
     auto data_encryptor =
         file_encryptor_ ? file_encryptor_->GetColumnDataEncryptor(path->ToDotString())
                         : nullptr;
-    auto ci_builder = page_index_builder_ && properties_->page_index_enabled(path) &&
-                              properties_->statistics_enabled(path)
+    auto ci_builder = page_index_builder_ && column_properties.page_index_enabled() &&
+                              column_properties.statistics_enabled()
                           ? page_index_builder_->GetColumnIndexBuilder(column_ordinal)
                           : nullptr;
-    auto oi_builder = page_index_builder_ && properties_->page_index_enabled(path)
+    auto oi_builder = page_index_builder_ && column_properties.page_index_enabled()
                           ? page_index_builder_->GetOffsetIndexBuilder(column_ordinal)
                           : nullptr;
-    auto codec_options = properties_->codec_options(path)
-                             ? properties_->codec_options(path).get()
-                             : nullptr;
-
-    std::unique_ptr<PageWriter> pager;
+    const CodecOptions* codec_options = column_properties.codec_options()
+                                            ? column_properties.codec_options().get()
+                                            : nullptr;
+    BloomFilter* bloom_filter =
+        bloom_filter_builder_ && column_properties.bloom_filter_enabled()
+            ? bloom_filter_builder_->GetOrCreateBloomFilter(column_ordinal)
+            : nullptr;
+
+    CodecOptions default_codec_options;
```

Comment:
Couldn't bear everytime we add an argument we need to edit it twice😅,so update this

Reply:
https://github.com/apache/arrow/pull/40768 do it here